### DISCUSSION TITLE
parallel_computation.py run mulltiple SU2 session

### DISCUSSION THREAD

**Discussion by GovoniT** [su2code/SU2#1633] [Q&A]
Hi, 

I have a small problem when running SU2 with parallel_computation.py :

Instead of launching one session on n procs, it launches n sessions...

I also try to launch the computation with mpirun, but the result is the same

OS : Ubuntu 22.04 LTS
mpi : mpich 4.0
SU2 : SU2-v7.3.1-linux64-mpi

**Comment by bigfooted** [comment]
does a normal mpi run work, like this:
`mpirun -n 2 SU2_CFD config.cfg `

Or do you have the same problem? It sounds like mpi is not set up correctly. Did you compile SU2, like this:
`./meson.py build -Dwith-mpi=enabled -Dcustom-mpi=true -Dextra-deps=mpich `

**Reply by GovoniT** [reply]
Thanks for the suggestion,

After recompiling SU2 with `./meson.py build -Dwith-mpi=enabled -Dcustom-mpi=true -Dextra-deps=mpich` and run the `./ninja -C build install`

I get the following errors :

> [1/801] Compiling C object externals/cgns/hdf5/libsu2hdf5.a.p/H5dbg.c.o
FAILED: externals/cgns/hdf5/libsu2hdf5.a.p/H5dbg.c.o 
cc -Iexternals/cgns/hdf5/libsu2hdf5.a.p -Iexternals/cgns/hdf5 -I../externals/cgns/hdf5 -fdiagnostics-color=always -std=c99 -O3 -fPIC -Wno-unused-result -Wno-unused-parameter -Wno-unused-variable -Wno-unused-but-set-variable -Wno-sign-compare -Wno-stringop-overflow -Wno-discarded-qualifiers -Wno-error=maybe-uninitialized -Wno-error=unused-function -Wno-error=unused-label -Wno-pedantic -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -D_POSIX_C_SOURCE=200809L -D_GNU_SOURCE -MD -MQ externals/cgns/hdf5/libsu2hdf5.a.p/H5dbg.c.o -MF externals/cgns/hdf5/libsu2hdf5.a.p/H5dbg.c.o.d -o externals/cgns/hdf5/libsu2hdf5.a.p/H5dbg.c.o -c ../externals/cgns/hdf5/H5dbg.c
In file included from ../externals/cgns/hdf5/H5private.h:27,
                 from ../externals/cgns/hdf5/H5dbg.c:33:
../externals/cgns/hdf5/H5public.h:63:10: fatal error: mpi.h: No such file or directory
   63 | #include <mpi.h>
      |          ^~~~~~~
compilation terminated.


it is like my mpi is not found but I think it is correctly installed on my machine
when I run `mpirun --version`

> mpirun (Open MPI) 4.1.2

and `which mpirun`
> /usr/bin/mpirun

**Reply by bigfooted** [reply]
You mentioned in your first post that you used mpich, but when you start mpirun, it says it's openmpi. So did you install both versions or only openmpi? If you want to use openmpi, then remove the custom-mpi and extra-deps arguments. 
I had some problems getting openmpi running on my ubuntu so I used mpich instead.

Another thing is that /usr/bin/mpirun is a symbolic link. In my case to another symbolic link, which then refers to mpich. Using this link you can switch between openmpi and mpich. If you have multiple mpi distributions, you have to make sure that you refer to the correct mpi.h with LD_LIBRARY_PATH

In the muon package manager, also check that you have installed the mpi-dev package, which contains the mpi.h files (for mpich, you have the mpich-dev package)

**Reply by GovoniT** [reply]
Indeed, there were several versions of mpi installed on my computer which must have caused the problem.

Many thanks for your help !
